stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewGroupDescriptor.java

```diff
@@ -0,0 +1,164 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+package com.facebook.stetho.inspector.elements.android;
+
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.facebook.stetho.common.Util;
+import com.facebook.stetho.inspector.elements.ChainedDescriptor;
+
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+final class ViewGroupDescriptor extends ChainedDescriptor<ViewGroup> {
+  // TODO: We're probably going to switch to another way of determining structural
+  //       changes in the View tree. So this gizmo with chaining OnHierarchyChangeListener
+  //       via reflection should go away soon.
+
+  private static final Field sOnHierarchyChangeListenerField;
+  static {
+    Field field;
+    try {
+      field = ViewGroup.class.getDeclaredField("mOnHierarchyChangeListener");
```

Comment:
I'm quite loathe to do this as well (the reflection hax0ring, that is). How about we leave as-is for now and yank it once diffing is in place? (which I'm definitely not treating as a low priority thing btw)